mypyc/transform/refcount.py

```diff
@@ -103,7 +103,7 @@ def transform_block(block: BasicBlock,
         key = (block, i)
 
         assert op not in pre_live[key]
-        dest = op.dest if isinstance(op, Assign) else op
+        dest = op.dest if isinstance(op, Assign) or isinstance(op, SetMem) else op
```

Comment:
I'm not sure if this is the right thing to do. SetMem doesn't modify the destination in a data-flow sense. In data-flow the destination is actually a source. 

(This may be related: It looks like the `type` attribute of `SetMem` is incorrect -- it should be void, and the `type` of the value being assigned needs to be stored in another attribute.)